sshd-common/src/main/java/org/apache/sshd/common/util/io/DirectoryScanner.java

```diff
@@ -115,49 +126,37 @@
  * @author <a href="mailto:levylambert@tiscali-dsl.de">Antoine Levy-Lambert</a>
  */
 public class DirectoryScanner {
-
     /**
      * The base directory to be scanned.
      */
-    protected File basedir;
+    private Path basedir;
 
     /**
      * The patterns for the files to be included.
      */
-    protected String[] includes;
-
-    /**
-     * The files which matched at least one include and no excludes
-     * and were selected.
-     */
-    protected List<String> filesIncluded;
+    private List<String> includePatterns;
 
     /**
-     * Whether or not the file system should be treated as a case sensitive
-     * one.
+     * Whether or not the file system should be treated as
+     * a case sensitive one.
      */
-    protected boolean isCaseSensitive = true;
+    private boolean caseSensitive = OsUtils.isUNIX();
```

Comment:
I am not sure if this would be true entirely. Apple's HFS is not case-sensitive. `isUNIX` just checks if it is Windows or not.

Reply:
The code is geared towards the 99.99% - if someone is running on top of HFS then they can override the case sensitivity

Reply:
I am trying to understand this better. You mean, overriding `caseSensitive` variable? This patch makes that `private`. Is that intentional?

Also, `DirectoryScanner` is a standalone class, it's not implementing an interface or extending a class. So, if someone has to override this variable and use, they have to override `ScpFileOpener#getMatchingFilesToSend` as well with the modified `DirectoryScanner` implementation. Is my understanding correct?

Reply:
I mean override `ScpFileOpener` and implement `ScpFileOpener#getMatchingFilesToSend` using a `DirectoryScanner` with some other case-sensitivity (NOTE: `DirectoryScanner#setCaseSensitivity` is available to override the default). This can be easily done when initializing the SCP client/server by setting a default `ScpFileOpener` when they are configured - thus no need to override the opener on each session.